packages/name-controller/src/NameController.ts

```diff
@@ -422,6 +448,7 @@ export class NameController extends BaseController<
         proposedNames: {},
         name: null,
         sourceId: null,
+        origin: NameOrigin.API,
```

Comment:
As this is handled in `setName`, do we need this here?

As you rightly pointed out, should the origin not be `null` initially until a name is set?

Reply:
Good catch.